ExposedAccessKeys/lambda_functions/lookup_cloudtrail_events.py

```diff
@@ -0,0 +1,65 @@
+import datetime
+import collections
+import boto3
+
+cloudtrail = boto3.client('cloudtrail')
+
+
+def lambda_handler(event, context):
+    account_id = event['account_id']
+    time_discovered = event['time_discovered']
+    username = event['username']
+    deleted_key = event['deleted_key']
+    exposed_location = event['exposed_location']
+    endtime = datetime.datetime.now()
+    interval = datetime.timedelta(hours=24)
+    starttime = endtime - interval
+    print('Retrieving events...')
+    events_pages = get_events_pages(username, starttime, endtime)
+    print('Summarizing events...')
+    event_names, resource_names, resource_types = get_events_summaries(events_pages)
+    return {
+        "account_id": account_id,
+        "time_discovered": time_discovered,
+        "username": username,
+        "deleted_key": deleted_key,
+        "exposed_location": exposed_location,
+        "event_names": event_names,
+        "resource_names": resource_names,
+        "resource_types": resource_types
+    }
+
+
+def get_events_pages(username, starttime, endtime):
+    try:
+        paginator = cloudtrail.get_paginator('lookup_events')
+        lookup_events_iterator = paginator.paginate(
```

Comment:
The intention of this summary is to give a quick view into the most recent API calls for the key. I agree that pagination may bog down the Lambda for a heavily used key. In this light I think this should be refactored to only return 1 page containing the most recent events. We can then update the summary message to say "This is a summary of the most recent API calls for this key. Please see CloudTrail for the full list of API calls." or something of the sort.